pkg/clusterconfig/clusterconfig.go

```diff
@@ -92,13 +102,59 @@ func Get() (*Config, error) {
 
 	cfg := &Config{}
 
-	cm, err := client.ConfigMaps(configNamespace).Get(configName, metaapi.GetOptions{})
-	if err != nil {
-		if errors.IsNotFound(err) {
-			logrus.Warnf("config-map %s/%s not found", configNamespace, configName)
-			return cfg, nil
+	cm, cmerr := client.ConfigMaps(configNamespace).Get(configName, metaapi.GetOptions{})
+
+	if cmerr != nil && errors.IsNotFound(cmerr) {
+		logrus.Warnf("config-map %s/%s not found, creating", configNamespace, configName)
+		cm = &corev1.ConfigMap{
+			TypeMeta: metav1.TypeMeta{
+				APIVersion: corev1.SchemeGroupVersion.String(),
+				Kind:       "ConfigMap",
+			},
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      configName,
+				Namespace: configNamespace,
+			},
+			Data: make(map[string]string),
+		}
+	}
+
+	awsCreds, err := client.Secrets(kubeSystemNamespace).Get(awsSecretName, metav1.GetOptions{})
+	if err != nil && errors.IsNotFound(err) {
+		logrus.Warnf("secret %s/%s not found, skipping aws configuration", kubeSystemNamespace, awsSecretName)
+	} else {
+		cm.Data["storage_type"] = string(regopapi.StorageTypeS3)
+		if v, ok := awsCreds.Data["aws_access_key_id"]; ok {
+			cm.Data["storage_aws_accesskey"] = string(v)
+		}
+		if v, ok := awsCreds.Data["aws_secret_access_key"]; ok {
+			cm.Data["storage_aws_secretkey"] = string(v)
+		}
+		clusterConfig, err := client.ConfigMaps(kubeSystemNamespace).Get(clusterConfigName, metav1.GetOptions{})
+		if err != nil {
+			return nil, err
+		} else {
+			installConfig := &installer.InstallConfig{}
+			if err := yaml.Unmarshal([]byte(clusterConfig.Data["install-config"]), &installConfig); err != nil {
+				return nil, err
+			}
+			cm.Data["storage_aws_region"] = installConfig.Platform.AWS.Region
+		}
+	}
+
+	// TODO: Insert other cloud configurations here
+	if errors.IsNotFound(cmerr) {
+		cm, err = client.ConfigMaps(configNamespace).Create(cm)
```

Comment:
I don't think this configmap serves any purpose.  @legionus introduced it as a mock version of the install-config. Now that we have the install-config, we should just be reading the install config and setting config fields on the registry resource+secret.  There is no reason to copy content from the install-config into a configmap that is consumed by the registry operator.

Reply:
10-4, I thought that looked kinda fishy when i was doing it :(